src/Common/tests/System/Net/Configuration.Certificates.cs

```diff
@@ -45,14 +45,24 @@ private static X509Certificate2Collection GetCertificateCollection(string certif
             {
                 // On Windows, .Net Core applications should not import PFX files in parallel to avoid a known system-level race condition.
                 // This bug results in corrupting the X509Certificate2 certificate state.
-                Assert.True(m.WaitOne(MutexTimeout), "Cannot acquire the global certificate mutex.");
-
-                var certCollection = new X509Certificate2Collection();
-                certCollection.Import(Path.Combine(TestDataFolder, certificateFileName), CertificatePassword, X509KeyStorageFlags.DefaultKeySet);
+                try
+                {
+                    Assert.True(m.WaitOne(MutexTimeout), "Cannot acquire the global certificate mutex.");
 
-                m.ReleaseMutex();
+                    var certCollection = new X509Certificate2Collection();
+                    certCollection.Import(Path.Combine(TestDataFolder, certificateFileName), CertificatePassword, X509KeyStorageFlags.DefaultKeySet);
```

Comment:
> I think that's because GC will still run the finalizer at some point before the process crashes.

I assume you're talking about the finally block, it seems odd to me that it can run before the catch block finishes executing. If I have code below the Debug.Fail in the catch block, I wouldn't expect that to run either, and so the finally block shouldn't run either otherwise the code would run in incorrect program order. Not sure what's happening here, but there are tests that verify abandoning global mutexes cross-process.